src/WebJobs.Script.WebHost/Security/FileSystemSecretsRepository.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.IO;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    /// <summary>
+    /// An <see cref="ISecretsRepository"/> implementation that uses the file system as the backing store.
+    /// </summary>
+    public sealed class FileSystemSecretsRepository : ISecretsRepository, IDisposable
+    {
+        private readonly string _secretsPath;
+        private readonly string _hostSecretsPath;
+        private readonly AutoRecoveringFileSystemWatcher _fileWatcher;
+        private bool _disposed = false;
+
+        public FileSystemSecretsRepository(string secretsPath)
+        {
+            if (secretsPath == null)
+            {
+                throw new ArgumentNullException(nameof(secretsPath));
+            }
+
+            _secretsPath = secretsPath;
+            _hostSecretsPath = Path.Combine(_secretsPath, ScriptConstants.HostMetadataFileName);
+
+            Directory.CreateDirectory(_secretsPath);
+
+            _fileWatcher = new AutoRecoveringFileSystemWatcher(_secretsPath, "*.json");
```

Comment:
This just pushes the concern down to the repository implementation. For files, we just keep doing what we were doing, for storage based implementations, we'll need to come up with an approach that makes sense.